src/client/utils/searchNarratives.ts

```diff
@@ -84,107 +113,110 @@ export default async function searchNarratives({
   sort = 'Recently updated',
   skip = 0,
   pageSize = 20,
-}: SearchParams): Promise<SearchResults> {
-  const options: SearchOptions = { query: { bool: { must: [] } }, pageSize };
-  // Query constraints for "must" conditions
-  const musts = [];
-  // Query constraints for "must not" conditions
-  const mustNots = [];
+}: SearchOptions): Promise<SearchResults> {
+  const params: SearchParams = {
+    types: ['KBaseNarrative.Narrative'],
+    paging: {
+      length: pageSize,
+      offset: skip,
+    },
+    track_total_hits: false,
+  };
   if (term) {
-    // Multi-match on narrative fields for the given search term
-    musts.push({ multi_match: { query: term, fields: SEARCH_FIELDS } });
+    params.search = {
+      query: term,
+      fields: ['agg_fields'], // Search on all text fields
+    };
   }
+  params.filters = {
+    operator: Operator.And,
+    fields: [{ field: 'is_temporary', term: false }],
+  };
   switch (category) {
     case 'own':
-      musts.push({ term: { creator: Runtime.username() } });
-      options.auth = true;
+      params.filters.fields.push({
+        field: 'creator',
+        term: Runtime.username(),
+      });
       break;
     case 'shared':
-      musts.push({ term: { shared_users: Runtime.username() } });
-      mustNots.push({ term: { creator: Runtime.username() } });
-      options.auth = true;
+      params.filters.fields.push({
+        field: 'creator',
+        not_term: Runtime.username(),
+      });
+      params.filters.fields.push({
+        field: 'shared_users',
+        term: Runtime.username(),
+      });
       break;
     case 'public':
-      musts.push({ term: { is_public: true } });
-      options.auth = false;
+      params.access = { only_public: true };
       break;
     case 'tutorials':
-      musts.push({ term: { is_public: true } });
-      options.auth = false;
-      musts.push({
-        bool: {
-          should: [
-            { match: { narrative_title: 'tutorial' } },
-            { match: { narrative_title: 'narratorial' } },
-          ],
-        },
-      });
+      params.access = { only_public: true };
+      params.filters.fields.push({ field: 'is_narratorial', term: true });
       break;
     default:
       throw new Error('Unknown search category');
   }
 
   if (sort) {
     if (sort === 'Recently created') {
-      options.sort = [{ creation_date: { order: 'desc' } }, '_score'];
+      params.sorts = [
+        ['creation_date', SortDir.Desc],
+        ['_score', SortDir.Desc],
```

Comment:
The ['_score', SortDir.Desc] seems common to all types. Any reason not to just start with that and push the specific one in the if statement?

So, like, 
```
if (sort) { 
    param.sorts = [['_score', SortDir.Desc]];
    if(sort === 'whatever') {
        params.sorts.push([other term])
    }
}
```

Reply:
Sure, will update :+1:

Reply:
This is updated